solvers.go

```diff
@@ -597,11 +597,11 @@ func (s *AdamSolver) Step(model []ValueGrad) (err error) {
 				return
 			}
 
-			if _, err = tensor.Mul(mHats, eta, tensor.UseUnsafe()); err != nil {
+			if _, err = tensor.Mul(mHats, negEta, tensor.UseUnsafe()); err != nil {
 				return errors.Wrap(err, pointWiseMulFail)
 			}
 
-			if _, err = tensor.Div(mHats, vHats, tensor.WithIncr(w)); err != nil {
```

Comment:
this was the problem

Reply:
wait why is this the problem?

Reply:
because the weights are added in the line 611 (couple of blocks below) and the formula says they are added once not twice 

![image](https://user-images.githubusercontent.com/73729/124474300-fe21de80-dd65-11eb-8bba-d8e865c02a95.png)